src/jdk.compiler/share/classes/com/sun/tools/javac/comp/Check.java

```diff
@@ -228,6 +234,12 @@ Lint setLint(Lint newLint) {
         return prev;
     }
 
+    boolean setImportSuppression(boolean newImportSuppression) {
+        boolean prev = importSuppression;
+        importSuppression = newImportSuppression;
```

Comment:
This is fine. Another alternative that I'm mentioning for completeness would have been for `attribImportType` to set a flag on AttrContext (and then augment some of the methods in `Check` to accept an extra `env` parameter).